internal/cluster/distribution/eks/eksprovider/workflow/update_asg_activity.go

```diff
@@ -238,6 +243,11 @@ func (a *UpdateAsgActivity) Execute(ctx context.Context, input UpdateAsgActivity
 			ParameterKey:     aws.String("NodeSecurityGroup"),
 			UsePreviousValue: aws.Bool(true),
 		},
+		sdkCloudformation.NewOptionalStackParameter(
+			"CustomNodeSecurityGroups",
+			input.SecurityGroups != nil,
+			strings.Join(input.SecurityGroups, ","),
+		),
```

Comment:
**Required**: this **MUST** be fixed.
1. This wouldn't work for updating old stack to new stack with no security groups specified, because there is no previous value.
2. In case of `EKS.ClusterAPI.UpdateCluster` the update should be overwrite-all not just overwrite-specified, so it has to be explicitly set which resolves the first problem (if the user specifies no security groups at the declarative request then the previous values should be cleared).

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/6688657f3edef416ad638ba0fde44a5bf1a109c4..5068a5af7e7afa7ed3cb28d19588a94cc18ef58a).